custom/enikshay/tasks.py

```diff
@@ -483,25 +484,34 @@ def get_aggregated_scores(self, latest_adherence_date, adherence_schedule_date_s
             end_date=update["aggregated_score_date_calculated"]
         )
 
+        doses_per_week = self.get_doses_per_week()
+        update['expected_doses_taken'] = int(((
+            (update['aggregated_score_date_calculated'] - adherence_schedule_date_start)).days / 7.0
```

Comment:
does this date range need to be inclusive? if so it should add 1

Reply:
This `expected_doses_taken` property is a count of all of the doses that should have been taken up until the date that cases are automatically closed (i.e. the "purge date"), or the date of the last adherence case, whichever is later. We don't want to count cases that remain open, as this would ruin the calculations on the phone. 

**aggregated_score_date_calculated = purge date**
- Today is `2017-01-31`. The purge_date is then `2017-01-01`
- Cases get closed at midnight of the 30th day after their `adherence_date` property here: https://enikshay.in/a/enikshay/data/edit/automatic_updates/edit/152/
    - If today is `2017-01-31`, then all cases where the adherence_date property is `2017-01-01` and earlier will be closed. i.e. all cases where `adherence_date` = `purge_date` are closed. 
- If the `adherence_schedule_date_start` is `2016-01-01`, then we want to count the number of days between `[2016-01-01, 2017-01-01]` = 366 (since 2016 was a leap year). 
- i.e. This should be inclusive. 

**aggregate_score_date_calculated = latest_adherence_date**
- Today is `2017-01-31`.
- The latest adherence case we received is `2016-12-31`
- The `adherence_schedule_date_start` is `2016-01-01`
- We want to count all the cases within the window [2016-01-01, 2016-12-31]. i.e. we also count the cases that occur on the `latest_adherence_date`. i.e. 31 days.
- Conclusion: this should also be inclusive.

Reply:
https://github.com/dimagi/commcare-hq/pull/18144